i18n/en/docusaurus-plugin-content-docs/current/integration/asset-workflow.addressables.md

```diff
@@ -0,0 +1,364 @@
+
+The following is an instruction on how to apply this feature.
+
+First, set up a build script to encrypt the assets as follows.
+
+- Create a ScriptableObject from the `Extreal>Integration.AssetWorkflow.Addressables.Editor>Encrypt Build Script` in the Create Asset menu.
+- Add the ScriptableObject you created to the `Build and Play Mode Scripts` from the `AddressablesAssetSettings` inspector.
+
+This setting allows you to choose `Build>New Build>Encrypt Build Script` in the `Addressables Group` window.
+
+Next, set up the ResourceProvider that will decrypt the asset.
+ICryptoStreamFactory must be implemented first.
+Implement ICryptoStreamFactory according to the requirements of your application.
+An example implementation using AES is shown below to show the implementation image.
+
+```csharp
+public class CryptoStreamFactory : ICryptoStreamFactory
+{
+    public CryptoStream CreateEncryptStream(Stream baseStream, AssetBundleRequestOptions options)
+        => CreateCryptoStream(baseStream, options, CryptoStreamMode.Write);
+
+    public CryptoStream CreateDecryptStream(Stream baseStream, AssetBundleRequestOptions options)
+        => CreateCryptoStream(baseStream, options, CryptoStreamMode.Read);
+
+    private static CryptoStream CreateCryptoStream
+    (
+        Stream baseStream,
+        AssetBundleRequestOptions options,
+        CryptoStreamMode mode
+    )
+    {
+        using var aes = CreateAesManaged(options);
+        var cryptor = mode == CryptoStreamMode.Write ? aes.CreateEncryptor() : aes.CreateDecryptor();
+        return new CryptoStream(baseStream, cryptor, mode);
+    }
+
+    [SuppressMessage("Usage", "CC0022")]
+    private static AesManaged CreateAesManaged(AssetBundleRequestOptions options)
+    {
+        const int keyLength = 128;
+        var salt = Encoding.UTF8.GetBytes(options.BundleName);
+
+        using var keyGen = new Rfc2898DeriveBytes(SecretVariables.CryptAssetPassword, salt, 100, HashAlgorithmName.SHA256);
+        using var ivGen = new Rfc2898DeriveBytes(SecretVariables.CryptAssetIv, salt, 1, HashAlgorithmName.SHA256);
+
+        var key = keyGen.GetBytes(keyLength / 8);
+        var iv = ivGen.GetBytes(keyLength / 8);
+
+        return new AesManaged
+        {
+            BlockSize = keyLength,
+            KeySize = keyLength,
+            Mode = CipherMode.CBC,
+            Padding = PaddingMode.PKCS7,
+            Key = key,
+            IV = iv
+        };
+    }
+}
+```
+
+Implement a class that extends CryptoAssetBundleProviderBase.
+Return a class that implements ICryptoStreamFactory.
+
+```csharp
+[DisplayName("Crypto AssetBundle Provider")]
+public class CryptoAssetBundleProvider : CryptoAssetBundleProviderBase
+{
+    public override ICryptoStreamFactory CryptoStreamFactory => new CryptoStreamFactory();
+}
+```
+
+Now you can select the class implemented in the `Asset Bundle Provider` from the inspector of the `Addressables Asset Group`.
```

Comment:
the class implemented in the `Asset Bundle Provider`
-> the implemented class as the `Asset Bundle Provider`

Reply:
Fixed!
https://github.com/extreal-dev/Extreal.Guide/pull/42/commits/46a5d9caaf19190a1628b4110e952f3ae7703baa